client/battle/CBattleInterfaceClasses.cpp

```diff
@@ -405,6 +405,12 @@ CBattleResultWindow::CBattleResultWindow(const BattleResult & br, CPlayerInterfa
 
 	exit = std::make_shared<CButton>(Point(384, 505), "iok6432.def", std::make_pair("", ""), [&](){ bExitf();}, SDLK_RETURN);
 	exit->setBorderColor(Colors::METALLIC_GOLD);
+	
+	if(allowReplay)
+	{
+		repeat = std::make_shared<CButton>(Point(24, 505), "icn6432.def", std::make_pair("", ""), [&](){ bRepeatf();}, SDLK_ESCAPE);

```

Comment:
Perhaps add description?
This is not H3 feature so it is not clear what "cancel" button does at battle result screen.
Proposition: add texts
"Replay battle" for tooltip in status bar
"Discard results of the battle and replay it manually. Only available once per battle" second string, for right-click popup. (feel free to adjust text to your liking)
These new strings can be added into config/translate.json